src/seedsigner/views/tools_views.py

```diff
@@ -241,6 +250,170 @@ def run(self):
+class ToolsCoinEntropyMnemonicLengthView(View):
+    def run(self):
+        # TRANSLATOR_NOTE: Inserts the number of coin flips needed for a 12-word mnemonic
+        twelve = _("12 words ({} flips)").format(mnemonic_generation.COIN__NUM_FLIPS__12WORD)
+        TWELVE = ButtonOption(twelve, return_data=mnemonic_generation.COIN__NUM_FLIPS__12WORD)
+
+        # TRANSLATOR_NOTE: Inserts the number of coin flips needed for a 24-word mnemonic
+        twenty_four = _("24 words ({} flips)").format(mnemonic_generation.COIN__NUM_FLIPS__24WORD)
+        TWENTY_FOUR = ButtonOption(twenty_four, return_data=mnemonic_generation.COIN__NUM_FLIPS__24WORD)
+
+        button_data = [TWELVE, TWENTY_FOUR]
+        selected_menu_num = ButtonListScreen(
+            title=_("Mnemonic Length"),
+            is_bottom_list=True,
+            is_button_text_centered=True,
+            button_data=button_data,
+        ).display()
+
+        if selected_menu_num == RET_CODE__BACK_BUTTON:
+            return Destination(BackStackView)
+
+        elif button_data[selected_menu_num] == TWELVE:
+            return Destination(ToolsCoin_Input_Method_View, view_args=dict(total_flips=mnemonic_generation.COIN__NUM_FLIPS__12WORD))
+
+        elif button_data[selected_menu_num] == TWENTY_FOUR:
+            return Destination(ToolsCoin_Input_Method_View, view_args=dict(total_flips=mnemonic_generation.COIN__NUM_FLIPS__24WORD))
+
+
+
+class ToolsCoin_Input_Method_View(View):
+    def __init__(self, total_flips: int):
+        super().__init__()
+        self.total_flips = total_flips
+
+    def run(self):
+        if self.total_flips == 128:
+            all_flips_text = _("128 coin flips in one go")
+            setwise_text = _("Sets of 11 coin flips")
+        elif self.total_flips == 256:
+            all_flips_text = _("256 coin flips in one go")
+            setwise_text = _("Sets of 11 coin flips")
+        else:
+            raise ValueError("Unsupported flip count")
+
+        ALL_FLIPS = ButtonOption(all_flips_text, return_data="all")
+        SETWISE = ButtonOption(setwise_text, return_data="setwise")
+
+        button_data = [ALL_FLIPS, SETWISE]
+        selected_menu_num = ButtonListScreen(
+            title=_("Input Method"),
+            is_bottom_list=True,
+            is_button_text_centered=True,
+            button_data=button_data,
+        ).display()
+
+        if selected_menu_num == RET_CODE__BACK_BUTTON:
+            return Destination(ToolsCoinEntropyMnemonicLengthView)
+
+        selected_option = button_data[selected_menu_num].return_data
+
+        if selected_option == "all":
+            return Destination(ToolsCoinEntropyEntryView, view_args={"total_flips": self.total_flips})
+
+        elif selected_option == "setwise":
+            return Destination(ToolsCoinEntropySetwiseEntryView, view_args={"total_flips": self.total_flips})
+
+
+
+class ToolsCoinEntropyEntryView(View):
```

Comment:
Since this implementation offers two options, it is always helpful to re-explain which option this `View` is for in a comment.